sdk/core/System.ClientModel/src/Pipeline/ClientPipeline.cs

```diff
@@ -144,6 +161,20 @@ public static ClientPipeline Create(
 
         int perTryIndex = index;
 
+        // Add logging policy just before the transport.
+
+        if (loggingOptions == null || loggingOptions.ShouldAddMessageLoggingPolicy())
+        {
+            if (loggingOptions == null || loggingOptions.ShouldUseDefaultMessageLoggingPolicy())
```

Comment:
I don't understand why these if statements need to be nested -- is there a way to simplify this logic to delegate the complexity of lines 166-176 to some type that can have the responsibility to e.g. determine whether/what logging policy to add to the pipeline?